akka-actor/src/main/scala/akka/pattern/Patterns.scala

```diff
@@ -347,9 +347,13 @@ object PatternsCS {
    *   },
    *   timeout);
    * }}}
+   *
+   * @param actor          the actor to be asked
+   * @param messageFactory function taking an actor ref and returning the message to be sent
+   * @param timeout        the timeout for the response before failing the returned completion stage
    */
-  def ask(actor: ActorRef, messageFactory: japi.Function[ActorRef, Any], timeout: Timeout): CompletionStage[AnyRef] =
-    scalaAsk(actor, messageFactory.apply _)(timeout).toJava.asInstanceOf[CompletionStage[AnyRef]]
+  def askWithReplyTo(actor: ActorRef, messageFactory: japi.function.Function[ActorRef, Any], timeout: Timeout): CompletionStage[AnyRef] =
+    akka.pattern.extended.ask(actor, messageFactory.apply(_))(timeout).toJava.asInstanceOf[CompletionStage[AnyRef]]
 
```

Comment:
Don't we have the same problem with the ones returning `Future`, i.e. in `Patterns` (still Java API)

Reply:
Yes, I think you are right, I'll investigate.